types/init.lua

```diff
@@ -368,7 +368,14 @@ local FontAttributes = {
 ---@field state "Charging" | "Discharging" | "Empty" | "Full" | "Unknown"
 
 ---@class WeztermPlugin
----@field require fun(url: string): any
+---@field require fun(url: string): any   -- 'require' function: takes a URL (string) and returns some result (any type)
+---@field list fun(): Plugin[]            -- 'list' function: returns an array of Plugin objects
+---@field update_all fun(): nil           -- 'update_all' function: performs updates, returns nothing
+
+---@class Plugin                          -- Represents the RepoSpec struct in Lua
+---@field url string                      -- The URL of the plugin repository
+---@field component string                -- The component or directory name of the plugin
+---@field plugin_dir string               -- The path to the plugin's directory
```

Comment:
It seems this `plugin` spec is still experimental? Not sure where you got the right typings for it.

Reply:
> Not sure where you got the right typings for it.

IIRC I just dug them up [somewhere in the source code](https://github.com/wez/wezterm/blob/main/lua-api-crates/plugin/src/lib.rs#L12-L16)